File: Back/NodeJS/src/infrastructure/BetModule.ts
import { ICommand } from "../application/Abstractions/Request/ICommand";
import { IBetModule } from "../application/Abstractions/IBetModule";
import { INotification } from "../application/Abstractions/Request/INotification";
import { IMediator } from "../infrastructure/Mediator";
import { IRequest } from "../application/Abstractions/Request/IRequest";

export class BetModule implements IBetModule {
    constructor(private mediator: IMediator) {}
    ExecuteNotification(notification: INotification): Promise<void> {
        return this.mediator.Publish(notification);
    }

    ExecuteCommand(command: ICommand): Promise<void> {
        return this.mediator.Send(command);
    }

    ExecuteQuery<T>(query: IRequest<T>): Promise<T> {
        return this.mediator.Send(query);
    }
}

File: Back/.Net/Modules/Bets/BetFriends.Bets.Infrastructure/BetModule.cs
using BetFriends.Bets.Application.Abstractions;
using BetFriends.Shared.Application.Abstractions.Messaging;
using MediatR;
using Microsoft.Extensions.DependencyInjection;

namespace BetFriends.Bets.Infrastructure;

public class BetModule : IBetModule
{
    public Task<T> ExecuteAsync<T>(IQuery<T> query)
    {
        using IServiceScope scope = BetCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        return mediator!.Send(query);
    }

    public Task ExecuteAsync(ICommand command)
    {
        using IServiceScope scope = BetCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        return mediator!.Send(command);
    }

    public Task ExecuteNotificationAsync(INotification notification)
    {
        using IServiceScope scope = BetCompositionRoot.BeginScope();
        var mediator = scope.ServiceProvider.GetService(typeof(IMediator)) as IMediator;
        return mediator!.Publish(notification);
    }
}
